server/src/database.rs

```diff
@@ -214,6 +212,37 @@ impl BlockDatabase {
         Ok(())
     }
 
+    /// Removes existing transaction from store
+    pub fn remove(&mut self, hash: &H256) -> Result<Option<Transaction>> {
+        fn search_transaction(hash: &H256, file: &fs::File, len: u64) -> Option<(Transaction, CursorState)> {
+            let mut it = TransactionsIterator::new(file, len, IteratorMode::ReadOnly);
+            while let Some(tx) = it.next() {
+                if tx.hash() == hash {
+                    return Some((tx, it.finish()));
+                }
+            }
+            None
+        };
+
+        let len = self.file.metadata()?.len();
+        self.file.seek(io::SeekFrom::Start(0))?;
+        if let Some((tx, CursorState { len, position, last_tx_len })) = search_transaction(hash, &self.file, len) {
+            // read the rest
+            let mut rest = Vec::with_capacity((len - position) as usize);
+            self.file.read_to_end(&mut rest)?;
+            // go back before the transaction
+            self.file.seek(io::SeekFrom::Start(position - last_tx_len))?;
+            // write back
+            self.file.write_all(&mut rest)?;
+            self.file.set_len(len - last_tx_len)?;
```

Comment:
what is exactly `len` `position` and `last_tx_len`? those name tell me nothing and I am unable to tell if the arithmetic here is correct or not

Reply:
My guess is:
- `position` is the end of the removed transaction in the file
- `len` is the length of the file
- `last_tx_len` is the length of the removed transaction

Reply:
also the len used here is not necessarily the length of the file, cause on line `229` the original local variable is shadowed